var/spack/repos/builtin/packages/wgrib2/package.py

```diff
@@ -3,29 +3,166 @@
 #
 # SPDX-License-Identifier: (Apache-2.0 OR MIT)
 
+import re
+
 from spack.package import *
 
 
-class Wgrib2(CMakePackage):
-    """The wgrib2 package functionality for interacting with, reading,
-    writing, and manipulating GRIB2 files."""
+class Wgrib2(MakefilePackage):
+    """Utility for interacting with GRIB2 files"""
 
     homepage = "https://www.cpc.ncep.noaa.gov/products/wesley/wgrib2"
-    url = "https://github.com/NOAA-EMC/NCEPLIBS-wgrib2/archive/refs/tags/v2.0.8-cmake-v6.tar.gz"
+    url = "https://www.ftp.cpc.ncep.noaa.gov/wd51we/wgrib2/wgrib2.tgz.v2.0.8"
+
+    maintainers = ["kgerheiser", "Hang-Lei-NOAA"]
+
+    version("3.1.1", sha256="9236f6afddad76d868c2cfdf5c4227f5bdda5e85ae40c18bafb37218e49bc04a", extension="tar.gz")
+    version("3.1.0", sha256="5757ef9016b19ae87491918e0853dce2d3616b14f8c42efe3b2f41219c16b78f", extension="tar.gz")
+    version("2.0.8", sha256="5e6a0d6807591aa2a190d35401606f7e903d5485719655aea1c4866cc2828160", extension="tar.gz")
+    version("2.0.7", sha256="d7f1a4f9872922c62b3c7818c022465532cca1f5666b75d3ac5735f0b2747793", extension="tar.gz")
+
+    variant("netcdf3", default=True,
+            description="Link in netcdf3 library to write netcdf3 files")
+    variant("netcdf4", default=False,
+            description="Link in netcdf4 library to write netcdf3/4 files")
```

Comment:
Can you build with both `netcdf3` and `netcdf4`?  If not, perhaps a multi-value variant like `ipolates` is order here.

Reply:
I don't think you can. I think you are suggesting something like this:
```
    variant("netcdf", default="3",
            description="Version of netCDF library to use",
            values=("3", "4"))
```
To me this would be confusing because to the inexperienced it reads like you can either use netcdf version 3 or version 4 files, the information that netcdf4 can read and write both 3 (classic) and 4 (all others) is lost?

Reply:
Do users actually still use netCDF3? You could change the description to something along the lines of:
```
description="Version of netCDF library to use. Note, version 4 also supports version 3 (classic).
```

Reply:
Oh, I like that!

Reply:
I added the conflict. But it turns out that the `+netcdf4` build is not working, because one needs to manually (or encoded in the `package.py` file) download the netcdf4 source code during the install. I'll demote the PR to draft mode and return it to "ready for review" once I got the time to fix this.